File: android/app/src/main/java/com/toursafe/modules/ServerApiModule.java
package com.toursafe.modules;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReadableMap;

public class ServerApiModule extends ReactContextBaseJavaModule {

    public ServerApiModule(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public String getName() {
        return "ServerApiModule";
    }

    @ReactMethod
    public void registerTourist(String aadhaarNumber, String passportNumber, String name, String tripItinerary, Promise promise) {
        try {
            // Implementation for tourist registration
            promise.resolve("Tourist registered successfully");
        } catch (Exception e) {
            promise.reject("REGISTRATION_ERROR", e.getMessage());
        }
    }

    @ReactMethod
    public void authenticateUser(String digitalId, String password, Promise promise) {
        try {
            // Implementation for user authentication
            promise.resolve("User authenticated successfully");
        } catch (Exception e) {
            promise.reject("AUTH_ERROR", e.getMessage());
        }
    }

    @ReactMethod
    public void createDigitalID(ReadableMap kycData, ReadableMap itinerary, Promise promise) {
        try {
            // Implementation for digital ID creation
            promise.resolve("Digital ID created successfully");
        } catch (Exception e) {
            promise.reject("DIGITAL_ID_ERROR", e.getMessage());
        }
    }

    @ReactMethod
    public void verifyDigitalID(String digitalIdHash, Promise promise) {
        try {
            // Implementation for digital ID verification
            promise.resolve("Digital ID verified successfully");
        } catch (Exception e) {
            promise.reject("VERIFICATION_ERROR", e.getMessage());
        }
    }

    @ReactMethod
    public void updateLocation(double latitude, double longitude, Promise promise) {
        try {
            // Implementation for location update
            promise.resolve("Location updated successfully");
        } catch (Exception e) {
            promise.reject("LOCATION_UPDATE_ERROR", e.getMessage());
        }
    }

    @ReactMethod
    public void getSafetyScore(double latitude, double longitude, Promise promise) {
        try {
            // Implementation for safety score calculation
            promise.resolve("Safety score retrieved successfully");
        } catch (Exception e) {
            promise.reject("SAFETY_SCORE_ERROR", e.getMessage());
        }
    }

    @ReactMethod
    public void reportEmergency(String emergencyType, double latitude, double longitude, String additionalData, Promise promise) {
        try {
            // Implementation for emergency reporting
            promise.resolve("Emergency reported successfully");
        } catch (Exception e) {
            promise.reject("EMERGENCY_ERROR", e.getMessage());
        }
    }

    @ReactMethod
    public void getGeofenceAlerts(double latitude, double longitude, Promise promise) {
        try {
            // Implementation for geofence alerts
            promise.resolve("Geofence alerts retrieved successfully");
        } catch (Exception e) {
            promise.reject("GEOFENCE_ERROR", e.getMessage());
        }
    }
}

File: android/app/src/main/java/com/toursafe/modules/ServerApiModule.kt
package com.toursafe.modules

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReadableMap

class ServerApiModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String {
        return "ServerApiModule"
    }

    @ReactMethod
    fun registerTourist(aadhaarNumber: String, passportNumber: String, name: String, tripItinerary: String, promise: Promise) {
        try {
            // Implementation for tourist registration
            promise.resolve("Tourist registered successfully")
        } catch (e: Exception) {
            promise.reject("REGISTRATION_ERROR", e.message)
        }
    }

    @ReactMethod
    fun authenticateUser(digitalId: String, password: String, promise: Promise) {
        try {
            // Implementation for user authentication
            promise.resolve("User authenticated successfully")
        } catch (e: Exception) {
            promise.reject("AUTH_ERROR", e.message)
        }
    }

    @ReactMethod
    fun createDigitalID(kycData: ReadableMap, itinerary: ReadableMap, promise: Promise) {
        try {
            // Implementation for digital ID creation
            promise.resolve("Digital ID created successfully")
        } catch (e: Exception) {
            promise.reject("DIGITAL_ID_ERROR", e.message)
        }
    }

    @ReactMethod
    fun verifyDigitalID(digitalIdHash: String, promise: Promise) {
        try {
            // Implementation for digital ID verification
            promise.resolve("Digital ID verified successfully")
        } catch (e: Exception) {
            promise.reject("VERIFICATION_ERROR", e.message)
        }
    }

    @ReactMethod
    fun updateLocation(latitude: Double, longitude: Double, promise: Promise) {
        try {
            // Implementation for location update
            promise.resolve("Location updated successfully")
        } catch (e: Exception) {
            promise.reject("LOCATION_UPDATE_ERROR", e.message)
        }
    }

    @ReactMethod
    fun getSafetyScore(latitude: Double, longitude: Double, promise: Promise) {
        try {
            // Implementation for safety score calculation
            promise.resolve("Safety score retrieved successfully")
        } catch (e: Exception) {
            promise.reject("SAFETY_SCORE_ERROR", e.message)
        }
    }

    @ReactMethod
    fun reportEmergency(emergencyType: String, latitude: Double, longitude: Double, additionalData: String, promise: Promise) {
        try {
            // Implementation for emergency reporting
            promise.resolve("Emergency reported successfully")
        } catch (e: Exception) {
            promise.reject("EMERGENCY_ERROR", e.message)
        }
    }

    @ReactMethod
    fun getGeofenceAlerts(latitude: Double, longitude: Double, promise: Promise) {
        try {
            // Implementation for geofence alerts
            promise.resolve("Geofence alerts retrieved successfully")
        } catch (e: Exception) {
            promise.reject("GEOFENCE_ERROR", e.message)
        }
    }
}